Bugzilla/Elastic/Role/Object.pm

```diff
@@ -9,7 +9,7 @@ package Bugzilla::Elastic::Role::Object;
 use 5.10.1;
 use Role::Tiny;
 
-requires qw(ES_TYPE ES_PROPERTIES es_document);
+requires qw(ES_TYPE ES_INDEX ES_SETTINGS ES_PROPERTIES es_document);
```

Comment:
Yep! The ChildObject role is what requires ES_PARENT_CLASS. A role is basically an interface that can provide default implementations.